### PR TITLE
Block coordinates on vm query

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: api/groups/baseVmValuesGroup.go [hunk 1]
```diff
@@ -136,11 +143,25 @@ func createSCQuery(request *VMValueRequest) (*data.SCQuery, error) {
...
+
 func returnBadRequest(context *gin.Context, errScope string, err error) {
 	message := fmt.Sprintf("%s: %s", errScope, err)
 	shared.RespondWith(context, http.StatusBadRequest, nil, message, data.ReturnCodeRequestError)
 }
 
-func returnOkResponse(context *gin.Context, dataToReturn interface{}) {
-	shared.RespondWith(context, http.StatusOK, gin.H{"data": dataToReturn}, "", data.ReturnCodeSuccess)
+func returnOkResponse(context *gin.Context, dataToReturn interface{}, blockInfo interface{}) {
+	shared.RespondWith(context, http.StatusOK, gin.H{"data": dataToReturn, "blockInfo": blockInfo}, "", data.ReturnCodeSuccess)
```
Comment (bogdan-rosianu) [inline api/groups/baseVmValuesGroup.go:166]: isn't `blockInfo` returned for 400-code responses as well?

File: process/scQueryProcessor.go [hunk 2]
```diff
@@ -36,27 +38,35 @@ func NewSCQueryProcessor(proc Processor, pubKeyConverter core.PubkeyConverter) (
...
 		request := scQueryProcessor.createRequestFromQuery(query)
 		response := &data.ResponseVmValue{}
 
-		httpStatus, err := scQueryProcessor.proc.CallPostRestEndPoint(observer.Address, SCQueryServicePath, request, response)
+		path := scQueryServicePath
+		if len(query.BlockHash) > 0 {
+			path = fmt.Sprintf("%s%s%s", path, blockHashSuffix, hex.EncodeToString(query.BlockHash))
+		}
+		if query.BlockNonce.HasValue {
+			path = fmt.Sprintf("%s%s%d", path, blockNonceSuffix, query.BlockNonce.Value)
```
Comment (bogdan-rosianu) [inline process/scQueryProcessor.go:66]: if a user provides both coordinates, the URL will look like this: 
`observer:port/vm-values/query?blockHash=hash?blockNonce=37`
refactor to make it work with both coordinates as well

File: process/scQueryProcessor_test.go [hunk 3]
```diff
@@ -112,19 +120,67 @@ func TestSCQueryProcessor_ExecuteQuery(t *testing.T) {
...
+	processor, _ := NewSCQueryProcessor(&mock.ProcessorStub{
+		ComputeShardIdCalled: func(addressBuff []byte) (u uint32, e error) {
+			return 0, nil
+		},
+		GetObserversCalled: func(shardId uint32) (observers []*data.NodeData, e error) {
+			return []*data.NodeData{
+				{Address: "adress1", ShardId: 0},
+			}, nil
+		},
+		CallPostRestEndPointCalled: func(address string, path string, dataValue interface{}, response interface{}) (int, error) {
```
Comment (bogdan-rosianu) [inline process/scQueryProcessor_test.go:158]: might assert the `address` + `path` as well

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: api/groups/baseVmValuesGroup.go (hunk 1 above)
Comment (sstanculeanu) [reply api/groups/baseVmValuesGroup.go:166]: from node, blockInfo is only returned on success

File: process/scQueryProcessor.go (hunk 2 above)
Comment (sstanculeanu) [reply process/scQueryProcessor.go:66]: fixed

File: process/scQueryProcessor.go [hunk 4]
```diff
@@ -59,11 +61,15 @@ func (scQueryProcessor *SCQueryProcessor) ExecuteQuery(query *data.SCQuery) (*vm
...
 
 		path := scQueryServicePath
-		if len(query.BlockHash) > 0 {
-			path = fmt.Sprintf("%s%s%s", path, blockHashSuffix, hex.EncodeToString(query.BlockHash))
-		}
 		if query.BlockNonce.HasValue {
-			path = fmt.Sprintf("%s%s%d", path, blockNonceSuffix, query.BlockNonce.Value)
+			path = fmt.Sprintf("%s%s%d", path, blockNonceParam, query.BlockNonce.Value)
+		}
+		if len(query.BlockHash) > 0 {
```
Comment (bogdan-rosianu) [inline process/scQueryProcessor.go:67]: could have been something like:
```
params := url.Values{}
if nonce {
   params.Add("blockNonce", fmt.Sprintf("%d", query.BlockNonce.Value))
}
if blockHash {
    params.Add("blockHash", query.BlockHash)
}

queryParams := param.Encode() // for ex: blockNonce=37&blockHash=aabbcc
```

File: process/scQueryProcessor_test.go [hunk 5]
```diff
@@ -141,23 +142,28 @@ func TestSCQueryProcessor_ExecuteQueryWithCoordinates(t *testing.T) {
...
-				{Address: "adress1", ShardId: 0},
+				{Address: providedAddr, ShardId: 0},
 			}, nil
 		},
 		CallPostRestEndPointCalled: func(address string, path string, dataValue interface{}, response interface{}) (int, error) {
-			require.True(t, strings.Contains(path, "blockNonce"))
+			require.Equal(t, providedAddr, address)
+			require.True(t, strings.Contains(path, "?blockNonce"))
 			require.True(t, strings.Contains(path, strconv.FormatUint(providedNonce, 10)))
+			require.True(t, strings.Contains(path, "&blockHash"))
```
Comment (bogdan-rosianu) [inline process/scQueryProcessor_test.go:165]: please add an assert for the entire path, not just `contains`

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**